### PR TITLE
Add a try catch in Log Task#run() to catch errors that occur and let LogCron continue planning for future logging.

### PR DESCRIPTION
**Pull request by reda-idtaleb**

### CODE REVIEW

File: back/src/main/java/com/ubikloadpack/jmeter/ulp/observability/task/LogTask.java [hunk 1]
```diff
@@ -52,9 +52,13 @@ public LogTask(
 	 */
 	@Override
 	public void run(long scheduledRunTimeMillis) {
-		LOG.debug("Sample buffer : {}",this.sampleQueue.size());
-		this.registry.logAndReset();
-		System.out.println(this.registry.guiLog());
+		try {
+			LOG.debug("Sample buffer : {}",this.sampleQueue.size());
```
Comment (pmouawad) [inline back/src/main/java/com/ubikloadpack/jmeter/ulp/observability/task/LogTask.java]: Add LOG.isDebugEnabled()

File: back/src/main/java/com/ubikloadpack/jmeter/ulp/observability/task/LogTask.java [hunk 2]
```diff
@@ -52,9 +52,13 @@ public LogTask(
...
+			this.registry.logAndReset();
+			System.out.println(this.registry.guiLog());
```
Comment (pmouawad) [inline back/src/main/java/com/ubikloadpack/jmeter/ulp/observability/task/LogTask.java]: Can you replace this with LOG.info

File: back/src/main/java/com/ubikloadpack/jmeter/ulp/observability/task/LogTask.java
Comment (pmouawad) [inline back/src/main/java/com/ubikloadpack/jmeter/ulp/observability/task/LogTask.java:1]: Can you rename the class so that it reflects its real responsability ?

File: back/src/main/java/com/ubikloadpack/jmeter/ulp/observability/task/LogTask.java (hunk 2 above)
Comment (reda-idtaleb) [reply back/src/main/java/com/ubikloadpack/jmeter/ulp/observability/task/LogTask.java]: ok

File: back/src/main/java/com/ubikloadpack/jmeter/ulp/observability/registry/MicrometerRegistry.java [hunk 3]
```diff
@@ -217,7 +217,7 @@ public SampleLog makeLog(String name, Date timestamp) {
 		Double totalThroughput = this.summaryRegistry.counter("count.total","sample",name).count() / (timeSinceFirstSampleCallInSeconds);
 		
 		return currentPeriodSummary == null ? null : new SampleLog(
-				Util.micrometerToOpenMetrics(name),
+				Util.makeOpenMetricsName(name),
```
Comment (pmouawad) [inline back/src/main/java/com/ubikloadpack/jmeter/ulp/observability/registry/MicrometerRegistry.java]: Is this related to the PR ? or should you rebase  your branch ?

File: back/src/main/java/com/ubikloadpack/jmeter/ulp/observability/listener/ULPObservabilityListener.java [hunk 4]
```diff
@@ -224,7 +224,7 @@ public void setRegex(String regex) {
 	}
 
 	public String getTotalLabel() {
-		return getPropertyAsString(ULPODefaultConfig.TOTAL_LABEL_PROP, ULPODefaultConfig.totalLabel());
+		return Util.makeOpenMetricsName(getPropertyAsString(ULPODefaultConfig.TOTAL_LABEL_PROP, ULPODefaultConfig.totalLabel()));
```
Comment (pmouawad) [inline back/src/main/java/com/ubikloadpack/jmeter/ulp/observability/listener/ULPObservabilityListener.java]: Is this related to the PR ? or should you rebase  your branch ?

File: back/src/main/java/com/ubikloadpack/jmeter/ulp/observability/task/LogTask.java
Comment (reda-idtaleb) [reply back/src/main/java/com/ubikloadpack/jmeter/ulp/observability/task/LogTask.java:1]: This is done in an other branch (#96_RI) because the renaming was related to the #96 issue.

File: back/src/main/java/com/ubikloadpack/jmeter/ulp/observability/registry/MicrometerRegistry.java (hunk 3 above)
Comment (reda-idtaleb) [reply back/src/main/java/com/ubikloadpack/jmeter/ulp/observability/registry/MicrometerRegistry.java]: Actually, this branch was already merged to the develop branch. And I commited in the same merged branch, then i did the PR . I shouldn't do that ? Should I create a new branch ?

File: back/src/main/java/com/ubikloadpack/jmeter/ulp/observability/listener/ULPObservabilityListener.java (hunk 4 above)
Comment (reda-idtaleb) [reply back/src/main/java/com/ubikloadpack/jmeter/ulp/observability/listener/ULPObservabilityListener.java]: The rebase is done. I took some times to understand what it do because i never used it. Now the PR should only has the new commits.

File: back/src/main/java/com/ubikloadpack/jmeter/ulp/observability/task/SampleMetricsLoggingTask.java [hunk 5]
```diff
@@ -52,9 +52,13 @@ public LogTask(
...
 	public void run(long scheduledRunTimeMillis) {
-		LOG.debug("Sample buffer : {}",this.sampleQueue.size());
-		this.registry.logAndReset();
-		System.out.println(this.registry.guiLog());
+		try {
+			LOG.info("Sample buffer : {}",this.sampleQueue.size());
+			this.registry.logAndReset();
+			System.out.println(this.registry.guiLog());
+		} catch (Exception e) {
+			LOG.error("Error while running the log task : {}", e);
```
Comment (pmouawad) [inline back/src/main/java/com/ubikloadpack/jmeter/ulp/observability/task/SampleMetricsLoggingTask.java]: Isn't there a missing parameter for {}

File: back/src/main/java/com/ubikloadpack/jmeter/ulp/observability/task/SampleMetricsLoggingTask.java [hunk 6]
```diff
@@ -52,9 +52,13 @@ public LogTask(
 	 */
 	@Override
 	public void run(long scheduledRunTimeMillis) {
-		LOG.debug("Sample buffer : {}",this.sampleQueue.size());
-		this.registry.logAndReset();
-		System.out.println(this.registry.guiLog());
+		try {
+			LOG.info("Sample buffer : {}",this.sampleQueue.size());
+			this.registry.logAndReset();
+			System.out.println(this.registry.guiLog());
```
Comment (pmouawad) [inline back/src/main/java/com/ubikloadpack/jmeter/ulp/observability/task/SampleMetricsLoggingTask.java]: Can you log as info instead of using System.out ?

File: back/src/main/java/com/ubikloadpack/jmeter/ulp/observability/task/SampleMetricsLoggingTask.java [hunk 7]
```diff
@@ -52,9 +52,13 @@ public LogTask(
 	 */
 	@Override
 	public void run(long scheduledRunTimeMillis) {
-		LOG.debug("Sample buffer : {}",this.sampleQueue.size());
-		this.registry.logAndReset();
-		System.out.println(this.registry.guiLog());
+		try {
+			LOG.info("Sample buffer : {}",this.sampleQueue.size());
+			this.registry.logAndReset();
```
Comment (pmouawad) [inline back/src/main/java/com/ubikloadpack/jmeter/ulp/observability/task/SampleMetricsLoggingTask.java]: Can you rename logAndReset to computeAndReset ?

**Review by pmouawad — CHANGES_REQUESTED** [review]
Thanks for PR, see my comments.

**Comment by reda-idtaleb** [comment]
> Thanks for PR, see my comments.

Done
